etl/steps/data/meadow/smoking/2024-05-30/cigarette_sales.py

```diff
@@ -0,0 +1,265 @@
+    "Germany",
+    "USSR and fSU",
+    "Armenia",
+    "Azerbaijan",
+    "Belarus",
+    "Georgia",
+    "Kazakhstan",
+    "Kyrgyzstan",
+    "Moldova",
+    "Russia",
+    "Tajikistan",
+    "Turkmenistan",
+    "Ukraine",
+    "Uzbekistan",
+    "Bosnia & Herzegovina",
+    "Macedonia",
+    "Federal Republic of Yugoslavia",
+]
+
+FORMAT_SPECIAL_CASES = {
+    "Armenia": {"header_row": 9, "cols": "A,B,C", "num_rows": 13},
+    "Azerbaijan": {"header_row": 9, "cols": "A,E,F", "num_rows": 13},
+    "Belarus": {"header_row": 9, "cols": "A,H,I", "num_rows": 13},
+    "Georgia": {"header_row": 9, "cols": "A,K,L", "num_rows": 14},
+    "Kazakhstan": {"header_row": 25, "cols": "A,B,C", "num_rows": 15},
+    "Kyrgyzstan": {"header_row": 25, "cols": "A,E,F", "num_rows": 13},
+    "Moldova": {"header_row": 25, "cols": "A,H,I", "num_rows": 14},
+    "Russia": {"header_row": 25, "cols": "A,K,L", "num_rows": 19},
+    "Tajikistan": {"header_row": 46, "cols": "A,B,C", "num_rows": 13},
+    "Turkmenistan": {"header_row": 46, "cols": "A,E,F", "num_rows": 13},
+    "Ukraine": {"header_row": 46, "cols": "A,H,I", "num_rows": 18},
+    "Uzbekistan": {"header_row": 46, "cols": "A,K,L", "num_rows": 15},
+    "Bosnia & Herzegovina": {"header_row": 9, "cols": "A,B,C", "num_rows": 3},
+    "Macedonia": {"header_row": 14, "cols": "A,B,C", "num_rows": 4},
+    "Federal Republic of Yugoslavia": {"header_row": 20, "cols": "A,B,C", "num_rows": 9},
+}
+
+NEW_COLUMNS = [
+    "year",
+    "manufactured_cigarettes_millions",
+    "manufactured_cigarettes_per_adult_per_day",
+    "handrolled_cigarettes_millions",
+    "handrolled_cigarettes_per_adult_per_day",
+    "total_cigarettes_millions",
+    "total_cigarettes_per_adult_per_day",
+    "all_tobacco_products_tonnes",
+    "all_tobacco_products_grams_per_adult_per_day",
+]
+
+ORIGINAL_COLUMNS = [
+    "Year",
+    "Manufactured cigarettes",
+    "Unnamed: 2",
+    "Unnamed: 3",
+    "Hand-rolled cigarettes",
+    "Unnamed: 5",
+    "Unnamed: 6",
+    "Total cigarettes",
+    "Unnamed: 8",
+    "Unnamed: 9",
+    "All tobacco products",
+    "Unnamed: 11",
+    "Unnamed: 12",
+]
+
+
+def is_not_string(value):
+    return not isinstance(value, str)
+
+
+def run(dest_dir: str) -> None:
+    #
+    # Load inputs.
+    #
+    # Retrieve snapshot.
+    snap = paths.load_snapshot("cigarette_sales.zip")
+    snap_meta = snap.to_table_metadata()
+
+    # Load data from snapshot.
+    zf = ZipFile(snap.path)
+    folder_name = zf.namelist()[0]
+
+    # Process data
+
+    # Load country tables from excel files one by one (to concatenate them later)
+    country_tables = []
+    # list needed to pad tables so they have the same format
+    addl_cols = [
+        "Unnamed: 3",
+        "Unnamed: 4",
+        "Unnamed: 5",
+        "Unnamed: 6",
+        "Unnamed: 7",
+        "Unnamed: 8",
+        "Unnamed: 9",
+        "Unnamed: 10",
+        "Unnamed: 11",
+        "Unnamed: 12",
+    ]
+
+    tb_from_excel = Table()
```

Comment:
I think this definition is not necessary, since you define tb_from_excel below on the first iteration.

Reply:
I do agree - I think the linter (Ruff?) complained when I tried to commit/ push it like that (I think since I assign tb_from_excel within the for loop that is not strictly guaranteed to be run, even if it always will be). I can figure out what exactly was the problem here in case we want to adjust the global linter config - but for now I would leave the definition here.

Reply:
So if i comment this line out and run "make test" in the console it fails with the following error:
/Users/tunaacisu/Code/etl/etl/steps/data/meadow/smoking/2024-05-30/cigarette_sales.py
  /Users/tunaacisu/Code/etl/etl/steps/data/meadow/smoking/2024-05-30/cigarette_sales.py:226:25 - error: "tb_from_excel" is possibly unbound (reportUnboundVariable)
1 error, 0 warnings, 0 informations 
Completed in 9.855sec
make: *** [check-typing] Error 1

If this is unwanted behavior, maybe this can be adjusted - but for now I found it easiest to just define tb_from_excel before the for-loop starts